Validation of Roman Numbers that start with "I"

JorgePatrick opened the pull request:
Implementation of the class RomanNumberValidator witch validates if the Roman Number is null, has a char that isnt a Roman Symbol and if a Roman Number starting with "I" is a valid Roman Number as well as the implementation of the unit test class for it

JorgePatrick committed:
Validation of Roman Numbers that start with "I"

Implementation of the class RomanNumberValidator witch validates if the Roman Number is null, has a char that isnt a Roman Symbol and if a Roman Number starting with "I" is a valid Roman Number as well as the implementation of the unit test class for it

JorgePatrick committed:
Remove .idea from repo

JorgePatrick committed:
Merge branch 'main' into input-validation

JorgePatrick committed:
Insert validation into NumberConverter class

Insert validation into NumberConverter class using dependency injection
Add mock RomanNumberValidator into NumberConverterTest
Add RomanNumberValidator into RomanCalculatorIntegrationTest

JorgePatrick committed:
Fix NumberConverterTest

Fix NumberConverterTest
Implemented validation tests in RomanCalculatorIntegrationTest

JorgePatrick commented on src/test/java/com/jorgepatrick/NumberConverterTest.java:
I've tried to mock RomanNumberValidator, but it didnt work, cause the method is void, I guess. I dont have a clue how to mock a class, that in the case of these tests is just not suposed to do nothing

JorgePatrick commented on src/test/java/com/jorgepatrick/RomanCalculatorIntegrationTest.java:
Not sure about how de setup method ended up

chicofaccin replied:
If its a method that returns a void, and you don't want to do anything inside the method, you don't need to mock the behaviour. The method will be skipped.

JorgePatrick commented on src/main/java/com/jorgepatrick/RomanNumberValidator.java:
Not sure if I've handled the exceptions as it should be @chicofaccin

chicofaccin commented on src/test/java/com/jorgepatrick/NumberConverterTest.java:
```diff
@@ -1,19 +1,32 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
 
 public class NumberConverterTest {
```
I think it didn't work because this annotation is missing: `@ExtendWith(MockitoExtension.class)`

chicofaccin replied:
This annotation will run the test with the Mockito context. You can try and debug and see the values of `RomanNumberValidator ` with and without `@ExtendWith(MockitoExtension.class)` to understand the difference.

JorgePatrick committed:
Validate numbers starting with V

Validate numbers starting with V
Change attribute name in RomanSymbols, so the method and attribute havent the same name
Correct exception tests to match the whole string

JorgePatrick committed:
Improve code

Cleaning code RomanNumberValidator

JorgePatrick commented on src/test/java/com/jorgepatrick/RomanCalculatorIntegrationTest.java:
```diff
@@ -5,20 +5,24 @@
+        romanNumberValidator = new RomanNumberValidator();
+        converter = new NumberConverter(romanNumberValidator);
+        romanCalculator = new RomanCalculator(converter);
     }
```
Is this setup well writen? @chicofaccin

JorgePatrick committed:
Update NumberConverterTest.java

No need to mock a void method
